lib/identity_validations/service_provider_validation.rb

```diff
@@ -34,11 +34,11 @@ def redirect_uris_are_valid
       return if redirect_uris.blank?
 
       redirect_uris.each do |uri|
-        next if !uri.include?('*') && (uri_valid?(uri) || uri_custom_scheme_only?(uri))
-
-        errors.add(:redirect_uris, :invalid)
-        break
+        errors.add(:redirect_uris, "#{uri} contains invalid wildcards(*)") if uri.include?('*')
+        errors.add(:redirect_uris, "#{uri} is not a valid URI") if !uri_valid?(uri) && !uri_custom_scheme_only?(uri)
```

Comment:
"blank is not a valid URI" isn't the best error message. Thoughts on wording to help partners decipher how to fix the url?

Reply:
Without adding a link (which isn't the best), the best we can do is call out the dev docs section about URIs or an RFC that describes valid URIs.